### DISCUSSION TITLE
Switch from arrow2 to arrow so we could upgrade datafusion

### DISCUSSION THREAD

**Discussion by evenyag** [GreptimeTeam/greptimedb#388] [General]
Now the arrow2 branch of datafusion is unmaintained, it would be better to switch to arrow so we could catch up the latest datafusion version.

**Comment by MichaelScofield** [comment]
Seems arrow2 forbid us from switching to stable rust?

```txt
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> /Users/luofucong/.cargo/registry/src/github.com-1ecc6299db9ec823/arrow2-0.10.1/src/lib.rs:8:39
  |
8 | #![cfg_attr(feature = "simd", feature(portable_simd))]
  |                                       ^^^^^^^^^^^^^
```

**Reply by evenyag** [reply]
This is probably caused by the `simd` feature of arrow2 requires nightly, we could disable this feature on stable rust.

**Reply by waynexia** [reply]
This is not only about nightly and stable. I have been blocked by arrow2 when I was going to bump nightly toolchain.

**Reply by evenyag** [reply]
Mainly due to the arrow2 branch of datafusion we used sticks to an old version of arrow2, thus we are also unable to upgrade arrow2 to a newer version.

**Comment by waynexia** [comment]
Tracked in https://github.com/GreptimeTeam/greptimedb/issues/555
